### ISSUE TITLE
using react native webrtc, Mic and camera error in android phone. Session error code: ERROR_CONTENT. Session error description: The BUNDLE group in answer contains a MID that was not in the offered group.. 

### ISSUE THREAD

**Issue by ashishgroomefy** [versatica/mediasoup-client#100]
**"react-native-webrtc": "^1.75.2",
"mediasoup-client": "^3.3.8",**

I am using react native webrtc in android smartphone. Using functions "enableMic, disableMic, enableWebcam, disableWebcam" from mediasoup demo. https://github.com/versatica/mediasoup-demo/blob/v3/app/lib/RoomClient.js

All theses functions are working fine but when i disablemic and then enablemic again, it shows this error
**"Session error code: ERROR_CONTENT. Session error description: The BUNDLE group in answer contains a MID that was not in the offered group.."**

Same error coming up with enabling cam after disabling cam doesnt works.

These errors are coming on when this piece of code is running.
this._micProducer = await this._sendTransport.produce( { track, codecOptions : { opusStereo : 1, opusDtx : 1 } });

this._webcamProducer = await this._sendTransport.produce( { track, encodings, codecOptions : { videoGoogleStartBitrate : 1000 } });

**When enabling mic this is the log----**
![Screenshot from 2019-12-18 17-49-49](https://user-images.githubusercontent.com/19470350/71086035-d3d9f180-21be-11ea-9024-c98318f9695b.png)

**when disabling mic** 
![Screenshot from 2019-12-18 17-50-51](https://user-images.githubusercontent.com/19470350/71086081-f23fed00-21be-11ea-9067-1334e162f5cc.png)

**when again enabling mic**
![Screenshot from 2019-12-18 17-51-31](https://user-images.githubusercontent.com/19470350/71086124-0d126180-21bf-11ea-9864-94af873f5442.png)

Same thing is happening with camera also.

**Comment by ibc** [comment]
I would need printed SDPs. You can get it from the logs.

**Comment by ashishgroomefy** [comment]
**Enabling Mic----**

```
 INFO  %cmediasoup-client:* %csend() **| calling pc.setLocalDescription() [offer:%o]**%c +54ms color: #00CC99 color: inherit {"sdp": "v=0
o=- 3040213337153279602 2 IN IP4 127.0.0.1
s=-
t=0 0
a=group:BUNDLE audio
a=msid-semantic: WMS 99e25faf-5304-4648-b809-51a04591fe5e
m=audio 9 UDP/TLS/RTP/SAVPF 111 103 104 9 102 0 8 106 105 13 110 112 113 126
c=IN IP4 0.0.0.0
a=rtcp:9 IN IP4 0.0.0.0
a=ice-ufrag:M6Ez
a=ice-pwd:ORDmPSv4MGdS+alDZEZLXQoz
a=ice-options:trickle renomination
a=fingerprint:sha-256 D6:2E:30:8D:A2:EF:66:4D:2C:67:A7:E4:96:90:16:03:9C:96:52:7F:6F:95:92:6D:36:1F:E5:E4:4D:7C:1F:B7
a=setup:actpass
a=mid:audio
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:2 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
a=sendonly
a=rtcp-mux
a=rtpmap:111 opus/48000/2
a=rtcp-fb:111 transport-cc
a=fmtp:111 minptime=10;useinbandfec=1
a=rtpmap:103 ISAC/16000
a=rtpmap:104 ISAC/32000
a=rtpmap:9 G722/8000
a=rtpmap:102 ILBC/8000
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:106 CN/32000
a=rtpmap:105 CN/16000
a=rtpmap:13 CN/8000
a=rtpmap:110 telephone-event/48000
a=rtpmap:112 telephone-event/32000
a=rtpmap:113 telephone-event/16000
a=rtpmap:126 telephone-event/8000
a=ssrc:2503766126 cname:pj5zSTRTvvIwR9LK
a=ssrc:2503766126 msid:99e25faf-5304-4648-b809-51a04591fe5e 2213a686-3496-4ca1-aba4-b067c990e87b
a=ssrc:2503766126 mslabel:99e25faf-5304-4648-b809-51a04591fe5e
a=ssrc:2503766126 label:2213a686-3496-4ca1-aba4-b067c990e87b
", "type": "offer"} color: #00CC99


 INFO  %cmediasoup-client:* %csend() | **calling pc.setRemoteDescription() [answer:%o**]%c +60ms color: #00CC99 color: inherit {"sdp": "v=0
o=mediasoup-client 10000 1 IN IP4 0.0.0.0
s=-
t=0 0
a=ice-lite
a=fingerprint:sha-512 58:AA:54:EF:D7:29:45:E2:2A:BF:07:A2:CB:17:50:AB:21:84:D5:06:EF:B0:36:0B:CD:20:10:02:C9:8D:C4:02:6E:EA:37:08:45:BA:2B:78:8B:1A:22:EF:B7:E0:2C:62:F0:2C:47:76:3F:84:1D:77:02:45:E2:B9:14:BE:40:91
a=msid-semantic: WMS *
a=group:BUNDLE audio
m=audio 7 UDP/TLS/RTP/SAVPF 111
c=IN IP4 127.0.0.1
a=rtpmap:111 opus/48000/2
a=fmtp:111 stereo=1;usedtx=1
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=setup:active
a=mid:audio
a=recvonly
a=ice-ufrag:smf5rtchy3ii6l76
a=ice-pwd:lb0ewmlev0fbjy03qolz5junsl00of1k
a=candidate:udpcandidate 1 udp 1076302079 192.168.1.90 46298 typ host
a=end-of-candidates
a=ice-options:renomination
a=rtcp-mux
a=rtcp-rsize
", "type": "answer"} color: #00CC99
 INFO  %cmediasoup-client:* %cconnection state changed to %s%c +140ms color: #00CC99 color: inherit connecting color: #00CC99
 INFO  %cmediasoup-client:* %cconnection state changed to %s%c +39ms color: #00CC99 color: inherit connected color: #00CC99
```

**-------Now Disabling MIC-------**

```
 LOG  onlcick disable cam/mic******
 INFO  disableMic()
 INFO  %cmediasoup-client:* %cclose()%c +0ms color: #00CC99 color: inherit color: #00CC99
 INFO  %cmediasoup-client:* %cstopSending() [localId:%s]%c +4m color: #00CC99 color: inherit 1 color: #00CC99
 INFO  %cmediasoup-client:* %cstopSending() | **calling pc.setLocalDescription() [offer:%o]**%c +181ms color: #00CC99 color: inherit {"sdp": "v=0
o=- 3040213337153279602 3 IN IP4 127.0.0.1
s=-
t=0 0
a=msid-semantic: WMS
m=audio 0 UDP/TLS/RTP/SAVPF 111 103 104 9 102 0 8 106 105 13 110 112 113 126
c=IN IP4 192.168.1.179
a=rtcp:9 IN IP4 0.0.0.0
a=candidate:2089671618 1 udp 2122260223 192.168.1.179 37597 typ host generation 0 network-id 5 network-cost 10
a=ice-ufrag:M6Ez
a=ice-pwd:ORDmPSv4MGdS+alDZEZLXQoz
a=ice-options:trickle renomination
a=fingerprint:sha-256 D6:2E:30:8D:A2:EF:66:4D:2C:67:A7:E4:96:90:16:03:9C:96:52:7F:6F:95:92:6D:36:1F:E5:E4:4D:7C:1F:B7
a=setup:actpass
a=mid:audio
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:2 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
a=inactive
a=rtcp-mux
a=rtpmap:111 opus/48000/2
a=rtcp-fb:111 transport-cc
a=fmtp:111 minptime=10;useinbandfec=1
a=rtpmap:103 ISAC/16000
a=rtpmap:104 ISAC/32000
a=rtpmap:9 G722/8000
a=rtpmap:102 ILBC/8000
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:106 CN/32000
a=rtpmap:105 CN/16000
a=rtpmap:13 CN/8000
a=rtpmap:110 telephone-event/48000
a=rtpmap:112 telephone-event/32000
a=rtpmap:113 telephone-event/16000
a=rtpmap:126 telephone-event/8000
", "type": "offer"} color: #00CC99
 **WARN  %**cmediasoup-client:WARN**:* %cstopSending() |** ignoring expected error due no sending tracks: %s%c +0ms color: #CC9933 color: inherit Failed to set local offer sdp: max-bundle is used but no bundle group found. color: #CC9933
```

**----Now Again Enabling MIC-----**

```
 LOG  onlcick enable cam/mic******
 INFO  enableMic()
 INFO  enableMic() | calling getUserMedia()
 INFO  %cmediasoup-client:* %cproduce() [track:%o]%c +6m color: #00CC99 color: inherit {"_enabled": true, "id": "5b48726a-b4da-45b2-b9a6-1beae9f56ba6", "kind": "audio", "label": "5b48726a-b4da-45b2-b9a6-1beae9f56ba6", "muted": false, "readonly": true, "readyState": "live", "remote": false} color: #00CC99
 INFO  %cmediasoup-client:* %csend() [kind:%s, track.id:%s]%c +2m color: #00CC99 color: inherit audio 5b48726a-b4da-45b2-b9a6-1beae9f56ba6 color: #00CC99
 INFO  %cmediasoup-client:* %csend() | calling pc.setLocalDescription() [offer:%o]%c +22ms color: #00CC99 color: inherit {"sdp": "v=0
o=- 3040213337153279602 4 IN IP4 127.0.0.1
s=-
t=0 0
a=group:BUNDLE audio
a=msid-semantic: WMS 99e25faf-5304-4648-b809-51a04591fe5e
m=audio 37597 UDP/TLS/RTP/SAVPF 111 103 104 9 102 0 8 106 105 13 110 112 113 126
c=IN IP4 192.168.1.179
a=rtcp:9 IN IP4 0.0.0.0
a=candidate:2089671618 1 udp 2122260223 192.168.1.179 37597 typ host generation 0 network-id 5 network-cost 10
a=ice-ufrag:M6Ez
a=ice-pwd:ORDmPSv4MGdS+alDZEZLXQoz
a=ice-options:trickle renomination
a=fingerprint:sha-256 D6:2E:30:8D:A2:EF:66:4D:2C:67:A7:E4:96:90:16:03:9C:96:52:7F:6F:95:92:6D:36:1F:E5:E4:4D:7C:1F:B7
a=setup:actpass
a=mid:audio
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:2 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
a=sendonly
a=rtcp-mux
a=rtpmap:111 opus/48000/2
a=rtcp-fb:111 transport-cc
a=fmtp:111 minptime=10;useinbandfec=1
a=rtpmap:103 ISAC/16000
a=rtpmap:104 ISAC/32000
a=rtpmap:9 G722/8000
a=rtpmap:102 ILBC/8000
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:106 CN/32000
a=rtpmap:105 CN/16000
a=rtpmap:13 CN/8000
a=rtpmap:110 telephone-event/48000
a=rtpmap:112 telephone-event/32000
a=rtpmap:113 telephone-event/16000
a=rtpmap:126 telephone-event/8000
a=ssrc:20110038 cname:pj5zSTRTvvIwR9LK
a=ssrc:20110038 msid:99e25faf-5304-4648-b809-51a04591fe5e 5b48726a-b4da-45b2-b9a6-1beae9f56ba6
a=ssrc:20110038 mslabel:99e25faf-5304-4648-b809-51a04591fe5e
a=ssrc:20110038 label:5b48726a-b4da-45b2-b9a6-1beae9f56ba6
", "type": "offer"} color: #00CC99
**ExceptionsManager.js:126 enableMic() | failed:"Session error code: ERROR_CONTENT. Session error description: max-bundle is used but no bundle group found.."**
```

**Comment by ibc** [comment]
Let's narrow the issue. It seems to happen when the **only** active m=section is closed. The PeerConnection produces (via `createOffer()`) a local SDP offer that later, when applied via `setLocalDescription()` makes react-native-webrtc or libwebrtc **complain** about the SDP offer generated by itself!

@saghul may I know which libwebrtc version does react-native-webrtc 1.75.2 use?

**Comment by ibc** [comment]
Ok, the good news is that I can reproduce the issue in Chrome 79 by forcing the mediasoup-client `ReactNative` handler in the mediasoup-demo.

**Comment by ibc** [comment]
Interesting, the issue does not happen if the `{ offerToReceiveAudio: false, offerToReceiveVideo: false }` options are not given to the PeerConnection constructor in the `ReactNative.js` handler, the issue does not happen. And it seems that those options are no longer needed since @saghul already remove these [ugly default values (set to true)](https://github.com/react-native-webrtc/react-native-webrtc/commit/d3544dd98c851f6a6c01038cc5a843a4c9a60195) in react-native-webrtc :)

**Comment by ibc** [comment]
@ashishgroomefy please checkout the `fix-react-native-pc-arguments` branch. It should fix the issue. And confirm, please, so I'll release a new version.

**Comment by ashishgroomefy** [comment]
> @ashishgroomefy please checkout the `fix-react-native-pc-arguments` branch. It should fix the issue. And confirm, please, so I'll release a new version.

Now it is working perfectly fine. Thank you @ibc

**Comment by ibc** [comment]
Great. Published in mediasoup-client 3.3.9.
